### PR TITLE
Add string join and title function

### PR DESCRIPTION
**Pull request by lixingwang**

### CODE REVIEW

File: function/string/descriptor.json [hunk 1]
```diff
@@ -283,6 +283,26 @@
         "type": "int"
       }
     },
+    {
+      "name": "join",
+      "description": "Convert array to string array and join concatenates the elements of items to create a single string. The separator string sep is placed between elements in the resulting string",
```
Comment (fm-tibco) [inline function/string/descriptor.json:288]: Maybe use the following `Converts an array to its string representation and concatenates its elements using the specified separator between each element.`

File: function/string/descriptor.json [hunk 2]
```diff
@@ -367,6 +387,20 @@
         "type": "string"
       }
     },
+    {
+      "name": "title",
```
Comment (fm-tibco) [inline function/string/descriptor.json:391]: Is this "toTitleCase"?  description : `convert the string to title case`

Comment (lixingwang) [reply function/string/descriptor.json:391]: Yes.  from Go it does `strings.Title()`

File: function/string/descriptor.json (hunk 1 above)
Comment (lixingwang) [reply function/string/descriptor.json]: done

File: function/string/descriptor.json (hunk 2 above)
Comment (fm-tibco) [reply function/string/descriptor.json:391]: @mellistibco, even though go uses .Title(), I noticed that we have .toLower(), .toUpper().  Should we use .toTitle() or something like that in this case?   strings.Title() doesn't seem very user friendly to me.

Comment (mellistibco) [reply function/string/descriptor.json:391]: I think we should stick with toLower, etc. As you said, Title isn’t very user friendly...

Comment (lixingwang) [reply function/string/descriptor.json:391]: @fm-tibco @mellistibco :  From Go.  `strings.Title` and `strings.ToTitle` are 2 different functions.  You can take a look https://golang.org/pkg/strings/#Title

Comment (fm-tibco) [reply function/string/descriptor.json:391]: go's toTitle isn't title case, its to Upper for everyone else.  We can make it "toTitleCase", title case is well known.

Comment (lixingwang) [reply function/string/descriptor.json]: @fm-tibco  Updated. please have another look

**Pull request merged**